Foundation/NSOperation.swift

```diff
@@ -81,17 +86,58 @@ public enum NSOperationQueuePriority : Int {
 
 public class NSBlockOperation : NSOperation {
     
-    public convenience init(block: () -> Void) {
+    private typealias ExecutionBlock = () -> ()
+    private var _executionBlocks = [ExecutionBlock]()
+    private let _executionBlocksLock = NSLock()
+    
+    public convenience init(block: () -> ()) {
+        self.init()
+        addExecutionBlock(block)
+    }
+    
+    public override func start() {
         NSUnimplemented()
     }
     
-    public func addExecutionBlock(block: () -> Void) {
+    public override func main() {
         NSUnimplemented()
     }
-
-    public var executionBlocks: [() -> Void] {
+    
+    public override var cancelled: Bool {
+        NSUnimplemented()
+    }
+    
+    public override func cancel() {
+        NSUnimplemented()
+    }
+    
+    public override var executing: Bool {
+        NSUnimplemented()
+    }
+    
+    public override var finished: Bool {
         NSUnimplemented()
     }
+    
+    public override var asynchronous: Bool {
+        NSUnimplemented()
+    }
+    
+    public override var ready: Bool {
+        NSUnimplemented()
+    }
+    
+    public func addExecutionBlock(block: () -> ()) {
+        _executionBlocksLock.lock()
```

Comment:
We cannot add a block if the operation is currently executing or has already finished. In this conditions maybe we should trigger a `fatalError` with a message.